sql/parse/parse.go

```diff
@@ -615,6 +615,14 @@ func selectToProjectOrGroupBy(se sqlparser.SelectExprs, g sqlparser.GroupBy, chi
 			return nil, err
 		}
 
+		agglen := len(selectExprs)
+		for i, ge := range groupingExprs {
+			// if GROUP BY index
+			if idx, err := strconv.Atoi(ge.String()); err == nil && idx > 0 && idx <= agglen {
```

Comment:
instead of this, you can check the ge is a `*expression.Literal` with integer type